### PR TITLE
Add documentation about Function.p.prototype

### LINKED ISSUE #1449: Where is the page for Function.prototype.prototype?
@ugultopu commented on [Tue Dec 01 2020](https://github.com/mdn/sprints/issues/3936)

The `prototype` instance property on a function instance allows that function to be used as a constructor. Although the `prototype` property is mentioned in the [new operator] page as:

> You can add a shared property to a previously defined object type by using the [Function.prototype] property. This defines a property that is shared by all objects created with that function, rather than by just one instance of the object type. The following code adds a color property with value "original color" to all objects of type Car, and then overwrites that value with the string "black" only in the instance object car1. For more information, see [prototype][Function.prototype].

Both links point to the same URL which goes to nowhere. That is, the [Function.prototype] page is not present and it simply redirects to the [Function] page.

This is a big omission if I'm not missing something.

[new operator]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new
[Function.prototype]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/prototype
[Function]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

Fix #1449 

I definitely think this is useful to have. I've wondered multiple times why this page doesn't exist!

#### Metadata

- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/function/prototype/index.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+title: Function.prototype.prototype
+slug: Web/JavaScript/Reference/Global_Objects/Function/prototype
+tags:
+  - Function
+  - JavaScript
+  - Property
+browser-compat: javascript.builtins.Function.prototype
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/function/prototype/index.md:8]:
```suggestion
spec-urls: https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-function-instances-prototype
```

Is that the right spec URL?

Anyway, I see a report of this flaw:

> **bad_bcd_queries**:
> 
> * `No BCD data for query: javascript.builtins.Function.prototype`

So, if we don’t have `javascript.builtins.Function.prototype` data in BCD, then I suggest we not include a `browser-compat` key until _after_ we get the data added in BCD — and so instead for now just use the `spec-urls` key to get the Specifications section populated at least.

File: files/en-us/web/javascript/reference/global_objects/function/prototype/index.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+## Specifications
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/function/prototype/index.md:108]:
```suggestion
## Browser compatibility

{{Compat}}

```

I suggest we entirely omit the “Browser compatibility” section until after we get the data for this into BCD.

**Review by sideshowbarker — CHANGES_REQUESTED** [review]

File: files/en-us/web/javascript/reference/global_objects/function/prototype/index.md (hunk 1 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/function/prototype/index.md:8]: Yes, this is the correct one.

RE: no BCD, I was planning to add BCD right before this is getting merged... But maybe we can move forward with `spec-urls` as well, considering BCD should be trivial (?)

Comment (sideshowbarker) [reply files/en-us/web/javascript/reference/global_objects/function/prototype/index.md]: > RE: no BCD, I was planning to add BCD right before this is getting merged... But maybe we can move forward with `spec-urls` as well, considering BCD should be trivial (?)

yeah, it’ll be easy enough to just do a follow-up PR here afterwards

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**